test/jackdaw/serdes/avro_test.clj

```diff
@@ -457,6 +457,199 @@
                                                              "topic"
                                                              (uuid/to-string uuid/+null+))))))))
 
+(deftest accept-unmangled-input
+  (let [schema {:name "testRecord"
+                :type "record"
+                :fields [{:name "string_field"
+                          :type "string"}
+                         {:name "long_field"
+                          :type "long"}
+                         {:name "optional_field"
+                          :type ["null" "int"]
+                          :default nil}
+                         {:name "nil_field"
+                          :type "null"}
+                         {:name "default_field"
+                          :type "long"
+                          :default 1}
+                         {:name "bytes_field"
+                          :type "bytes"}
+                         {:name "enum_field"
+                          :type {:type "enum"
+                                 :name "weird_values"
+                                 :symbols ["a_1" "B3"]}}
+                         {:name "map_field"
+                          :type ["null" {:type "map"
+                                         :values bananas-schema}]}
+                         {:name "array_field"
+                          :type ["null" {:name "subrecords"
+                                         :type "array"
+                                         :items "banana"}]}
+                         {:name "uuid_field"
+                          :type {:type "string",
+                                 :logicalType "uuid"}}]}
+        schema-str (json/write-str schema)
+        serde      (->serde schema-str)]
+
+    (is (= (round-trip serde "bananas"
```

Comment:
It took me a while before I realized that the difference between the 1st and 2nd round-trip tests here was that the enum field was expressed as a string in the first, and a keyword in the second. Would you mind putting each one inside a `testing` block so that they can more easily be distinguished?

Reply:
Thanks. I have restructured the tests and added documentation to make them more readable. 

I used a dynamic var for `*mangle-names* ` as it made more sense for me to have it as an 'application-level configuration'. It will lead to a consistent low level design within an application  - either use idiomised values or not. (although, I agree the degree of ambiguity wouldn't be much as there will always be a default setting). 

I noticed that the mangling is done only for the serde and not for json etc., So, as you mentioned, it would make sense to drop it altogether and push the responsibility to the apps if they need such a feature. The dynamic var will be a nice way to provide backward compatibility during transition if you decide to drop mangling in a future release.

Let me know your thoughts.

Reply:
Hi @cddr, @creese any thoughts on this PR ? Let me know if you still like the dyn var to be moved to the `topic-config` map (last arg to the serde function) instead. The map is named `topic-config` which makes it a weird place for the `mangle-names` key - what do you think ?